protocols/mdns/src/behaviour.rs

```diff
@@ -58,17 +60,26 @@ pub struct MdnsConfig {
     /// peer joins the network. Receiving an mdns packet resets the timer
     /// preventing unnecessary traffic.
     pub query_interval: Duration,
+    /// IP version type.
+    pub domain: IPType,
 }
 
 impl Default for MdnsConfig {
     fn default() -> Self {
         Self {
             ttl: Duration::from_secs(6 * 60),
             query_interval: Duration::from_secs(5 * 60),
+            domain: IPType::IPv4,
```

Comment:
I don't think we need this config, we can just work with ip6 in addition to ip4

Reply:
You can add a multicast_addr: std::net::IpAddr instead and just combine that with a port to get a socketaddr

Reply:
done